egs/wsj/s5/steps/nnet3/chain2/process_egs.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+
+# Copyright   2019  Johns Hopkins University (Author: Daniel Povey).  Apache 2.0.
+# Copyright   2019  Idiap Research Institute (Author: Srikanth Madikeri).  Apache 2.0.
+#
+# This script takes nnet examples dumped by steps/chain/get_raw_egs.sh and
+# combines the chunks into groups by speaker (to the extent possible; it may
+# need to combine speakers in some cases), locally randomizes the result, and
+# dumps the resulting egs to disk.  Chunks of these will later be globally
+# randomized (at the scp level) by steps/chaina/randomize_egs.sh
+
+
+# Begin configuration section.
+cmd=run.pl
+chunks_per_group=4
+num_repeats=2  # number of times we repeat the same chunks with different
+               # grouping.  Recommend 1 or 2; must divide chunks_per_group
+compress=true   # set this to false to disable compression (e.g. if you want to see whether
+                # results are affected).
+
+num_utts_subset=300     # number of utterances in validation and training
+                        # subsets used for shrinkage and diagnostics.
+
+
+shuffle_buffer_size=5000   # Size of buffer (containing grouped egs) to use
+                           # for random shuffle.
+
+stage=0
+nj=5             # the number of parallel jobs to run.
+srand=0
+
+echo "$0 $@"  # Print the command line for logging
+
+if [ -f path.sh ]; then . ./path.sh; fi
+. parse_options.sh || exit 1;
+
+
+if [ $# != 2 ]; then
+  echo "Usage: $0 [opts] <raw-egs-dir> <processed-egs-dir>"
+  echo " e.g.: $0 --chunks-per-group 4 exp/chaina/tdnn1a_sp/raw_egs exp/chaina/tdnn1a_sp/processed_egs"
```

Comment:
If I understand correctly, the chunks-per-group actually does nothing,and neither does num-repeats.  Can you make sure they are completely removed, including from the diagnostic part below?  (There is an awk script).